src/Tezos/Address.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+
+#include "AddressChecksum.h"
+#include "../Hash.h"
+#include "../HexCoding.h"
+
+#include <TrezorCrypto/sha3.h>
+
+#include <cassert>
+
+using namespace TW::Tezos;
+
+bool Address::isValid(const std::string& string) {
+  // TODO: implement
```

Comment:
Addresses are valid if they begin with `tz1`, `tz2` or `tz3` and are base58Check encoded. 

There's a Base58Check functionality provided in TrustCore which you can use.

Here's some test cases:
- Valid:
tz1eZwq8b5cvE2bPKokatLkVMzkxz24z3Don
tz2Rh3NYeLxrqTuvaZJmaMiVMqCajeXMWtYo
tz3RDC3Jdn4j15J7bBHZd29EUee9gVB1CxD9

Invalid: 
NmH7tmeJUmHcncBDvpr7aJNEBk7rp5zYsB1qt // Invalid prefix, valid checksum
tz1eZwq8b5cvE2bPKokatLkVMzkxz24z3AAAA // Valid prefix, invalid checksum
1tzeZwq8b5cvE2bPKokatLkVMzkxz24zAAAAA  // Invalid prefix, invalid checksum